From 3d776f568720d79805b5176bcbac8fbb5b38998c Mon Sep 17 00:00:00 2001
From: Broyojo <dha@xoba.com>
Date: Mon, 5 Oct 2026 21:32:07 -0400
Subject: [PATCH] wine.inf: Update the VC runtime version to 14.51.36247.

Unreal Engine's launcher compares the version in
HKLM\Software\Microsoft\VisualStudio\14.0\VC\Runtimes\<arch> with the
toolset the game was built with, and runs the game's redistributable
installer when the registered one is older. 14.42.34433 dates from
November 2024, so a game built since then starts that installer on every
prefix; Astroneer's (UE4PrereqSetup, which goes on to install .NET
Framework 4) does not finish. 14.51.36247 is what the redistributable
Steam installs today registers.
---
 loader/wine.inf.in | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 965d3b261c..e795a8deae 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -663,28 +663,28 @@ HKLM,%CurrentVersionNT%\Ports,"COM3:",2,"9600,n,8,1"
 HKLM,%CurrentVersionNT%\Ports,"COM4:",2,"9600,n,8,1"
 
 [VCRuntime.ntamd64]
-HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\x64,"Bld",0x10003,0x00008681
+HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\x64,"Bld",0x10003,0x00008d97
 HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\x64,"Installed",0x10003,0x00000001
 HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\x64,"Major",0x10003,0x0000000e
-HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\x64,"Minor",0x10003,0x0000002a
+HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\x64,"Minor",0x10003,0x00000033
 HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\x64,"Rbld",0x10003,0x00000000
-HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\x64,"Version",2,"14.42.34433.0"
+HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\x64,"Version",2,"14.51.36247.0"
 
 [VCRuntime.ntarm64]
-HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\arm64,"Bld",0x10003,0x00008681
+HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\arm64,"Bld",0x10003,0x00008d97
 HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\arm64,"Installed",0x10003,0x00000001
 HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\arm64,"Major",0x10003,0x0000000e
-HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\arm64,"Minor",0x10003,0x0000002a
+HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\arm64,"Minor",0x10003,0x00000033
 HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\arm64,"Rbld",0x10003,0x00000000
-HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\arm64,"Version",2,"14.42.34433.0"
+HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\arm64,"Version",2,"14.51.36247.0"
 
 [VCRuntime.ntx86]
-HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\x86,"Bld",0x10003,0x00008681
+HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\x86,"Bld",0x10003,0x00008d97
 HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\x86,"Installed",0x10003,0x00000001
 HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\x86,"Major",0x10003,0x0000000e
-HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\x86,"Minor",0x10003,0x0000002a
+HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\x86,"Minor",0x10003,0x00000033
 HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\x86,"Rbld",0x10003,0x00000000
-HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\x86,"Version",2,"14.42.34433.0"
+HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\x86,"Version",2,"14.51.36247.0"
 
 [RegisterDllsSection]
 ;;some dlls have to be registered first
-- 
2.54.0 (Apple Git-157)

